deeplake/core/dataset/deeplake_query_dataset.py

```diff
@@ -0,0 +1,375 @@
+        
+        Examples:
+
+            Creating a simple dataloader object which returns a batch of numpy arrays
+
+            >>> import deeplake
+            >>> ds_train = deeplake.load('hub://activeloop/fashion-mnist-train')
+            >>> train_loader = ds_train.dataloader().numpy()
+            >>> for i, data in enumerate(train_loader):
+            ...     # custom logic on data
+            ...     pass
+
+
+            Creating dataloader with custom transformation and batch size
+
+            >>> import deeplake
+            >>> import torch
+            >>> from torchvision import datasets, transforms, models
+            >>> 
+            >>> ds_train = deeplake.load('hub://activeloop/fashion-mnist-train')
+            >>> tform = transforms.Compose([
+            ...     transforms.ToPILImage(), # Must convert to PIL image for subsequent operations to run
+            ...     transforms.RandomRotation(20), # Image augmentation
+            ...     transforms.ToTensor(), # Must convert to pytorch tensor for subsequent operations to run
+            ...     transforms.Normalize([0.5], [0.5]),
+            ... ])
+            ...
+            >>> batch_size = 32
+            >>> # create dataloader by chaining with transform function and batch size and returns batch of pytorch tensors
+            >>> train_loader = ds_train.dataloader()\\
+            ...     .transform({'images': tform, 'labels': None})\\
+            ...     .batch(batch_size)\\
+            ...     .shuffle()\\
+            ...     .pytorch()
+            ...
+            >>> # loop over the elements
+            >>> for i, data in enumerate(train_loader):
+            ...     # custom logic on data
+            ...     pass
+
+            Creating dataloader and chaining with query
+
+            >>> ds = deeplake.load('hub://activeloop/coco-train')
+            >>> train_loader = ds_train.dataloader()\\
+            ...     .query("(select * where contains(categories, 'car') limit 1000) union (select * where contains(categories, 'motorcycle') limit 1000)")\\
+            ...     .pytorch()
+            ...
+            >>> # loop over the elements
+            >>> for i, data in enumerate(train_loader):
+            ...     # custom logic on data
+            ...     pass
+
+        **Restrictions**
+
+        The new high performance C++ dataloader is part of our Growth and Enterprise Plan .
+
+        - Users of our Community plan can create dataloaders on Activeloop datasets ("hub://activeloop/..." datasets).
+        - To run queries on your own datasets, `upgrade your organization's plan <https://www.activeloop.ai/pricing/>`_.
+        """
+        from deeplake.enterprise import DeepLakeDataLoader
+
+        dataloader = DeepLakeDataLoader(self, _indra_dataset=self.indra_ds)
+        return dataloader
+
+    @property
+    def no_view_dataset(self):
+        return self
```

Comment:
This doesnt do anything?

Reply:
This is used during shuffling, to get initial dataset and apply schedule on that. Returning self because for non-linear view there's no need to rely on initial dataset chunk structure.